api/REST-API.md

```diff
@@ -1,191 +1,574 @@
 
+```json
+[{"id":1,"first_name":"Fremont","last_name":"Broader","email":"fbroader0@bloglines.com","gender":"Male","ip_address":"23.27.246.1"},{"id":2,"first_name":"Georgetta","last_name":"Blamey","email":"gblamey1@wisc.edu","gender":"Female","ip_address":"17.110.6.159"},{"id":3,"first_name":"Meghann","last_name":"Quillinane","email":"mquillinane2@surveymonkey.com","gender":"Female","ip_address":"237.88.226.148"},{"id":4,"first_name":"Kerby","last_name":"Mate","email":"kmate3@si.edu","gender":"Male","ip_address":"81.44.87.187"},{"id":5,"first_name":"Loren","last_name":"Brabon","email":"lbrabon4@umich.edu","gender":"Female","ip_address":"47.137.187.14"}]
 ```
-HTTP/1.1 200 OK
 
-{
-   "result": "FAIL",
-   "errors": ["Unable to find event with ID 123"],
-   "data": null
-}
+So, the same way, you can setup the second GET method, as planned:
+
+|Route|HTTP Method|
+|-|-|
+|`/invoices`|`GET`|
+
+With Express, routing is as easy as chaining the HTTP method to the `route()` method, taking care of putting the route you want to lookup inside of the `route()` method parenthesis.
+
+```javascript
+app
+  .route('/invoices')
+  .get(
+    (req, res) => {
+      res.status(200).send(invoices);
+    }
+  )
 ```
+This time, if you test, you will get the entire invoices collection.
+
+However, the beauty of a RESTful API is that we want to use the possibility to fetch a particular customer or invoice. To do so, we need to create another set of routing that includes the ID of the record we wish to inquire about. For example, if we look at our data, we know that there is a customer with id 1 and an invoice with id 6. If we go back to our plan, we can see that we already had planned the route for these:
+
+|Route|HTTP Method|
+|-|-|
+|`/customers/{id}`|`GET`|
+|`/invoices/{id}`|`GET`|
 
-#### HTTP Status Codes
+So, to create these, simply add another routing to the `app`, but this time we need to fetch the ID that's included in the parameters. If you recall your server lessons, this can be done by accessing the `req.params` object. We then should iterate through all records until we find the record that has the same `id` because that is the property that we are looking for. Of course, you could use any of the object's properties, here. You would just need to adjust the route appropriately, but we will not cover this for now.
 
-But again HTTP already designed a perfectly useful standard of status codes, so let's use them.
+```javascript
+app
+  .route('/customers/:id')
+  .get((req, res) => {
+    let customer_id = req.params.id;
+    let status = 400;
+    let response = "Unable to fetch data!";
+    customers.forEach((customer) => {
+              status = 200;
+        response = customer;
+      }
+    });
+    res.status(status).send(response);
+  })
+
+app
+  .route('/invoices/:id')
+  .get((req, res) => {
+    let invoice_id = req.params.id;
+    let status = 400;
+    let response = "Unable to fetch data!";
+    invoices.forEach((invoice) => {
```

Comment:
Again, not sure why an invoices/:id route would return an array of invoices. If it did, wouldn’t you need to filter it at some point?

Reply:
Same than above. I apologize for the confusion.

Reply:
Fixed in [e76944c](https://github.com/Techtonica/curriculum/pull/1410/commits/e76944c2ebe4ae93d9adf1edab007af9e6589c7b)